mongoid.js

```diff
@@ -109,7 +109,12 @@ MongoId.prototype.mongoid = MongoId.prototype.fetch;
 var _zeroPadding = ["", "0", "00", "000", "0000", "00000", "000000", "0000000"];
 function hexFormat(n, width) {
     var s = n.toString(16);
-    return _zeroPadding[width - s.length] + s;
+    var paddingNeeded = width - s.length;
+    if (paddingNeeded > 0) {
+        return _zeroPadding[width - s.length] + s;
+    } else {
+        return s.substr(0, width);
```

Comment:
for most id fields keeping the least significant bits would preserve uniqueness better (ie, for all the incrementing fields like process id), so maybe `return s.substr(-width)`.  In this case it is the process id that was too long.

Note that with a clipped process id it's not possible to guarantee uniqueness (well, one could formulate a "machine id" that preserved the process uniqueness I suppose)